### DISCUSSION TITLE
How to get IntelliSense working on Uno.Shared?

### DISCUSSION THREAD

**Discussion by HubKing** [unoplatform/uno#7982] [Q&A]
In the Uno solution that was created by the VS 2022 extension, there is the Uno.Shared project. It has `MainPage.xaml`. If I open it, unlike the same file in a WPF prject, I see no visual designer. That is understandable, since this `MainPage.xaml` is neither WPF nor UWP, but Uno's own UI framework. But there is not even any IntelliSense. That is, if I type `<B`, I do not see a list that shows `Button`. This makes it really difficult to create the UI, because I have to lookup [the web documentation for Uno](https://platform.uno/docs/articles/implemented-views.html).

Is IntelliSense not implemented for Uno XAML, or is this a bug of VS 2022?

**Comment by jeromelaban** [comment]
Thanks for the report. To get intellisense, [you'll need to select UWP](https://platform.uno/docs/articles/get-started-wizard.html#the-xaml-editor-shows-the-type-page-does-not-support-direct-content-message) on the top left of the editor.

As for the intellisense about unsupported views, this is not yet supported. As a workaround, you can write the type in the cs code-behind file, you'll get green underlining that'll tell you if something is supported or not.

**Reply by davidbuckleyni** [reply]
This option is now misssing in lattest vs 2022
Even if I select Windows it does not work I see no UWP option.

![image](https://user-images.githubusercontent.com/5619229/176224966-bbbd9c87-5d34-4f7e-9c7b-9e7ecc0df15b.png)

![image](https://user-images.githubusercontent.com/5619229/176225128-da31c2b7-ab87-4ac2-9519-aa7ffd0570ba.png)

![image](https://user-images.githubusercontent.com/5619229/176225213-fbe7c57f-6a0c-4cc4-bd49-3df86810133d.png)

**Reply by jeromelaban** [reply]
It's expected that you'll only see `.Windows`. Depending on the template you chose, you'll see either `.UWP` or `.Windows`.

**Reply by davidbuckleyni** [reply]
But that does not answer the question if i pick windows the intellensense is still a bit broken

**Reply by jeromelaban** [reply]
Can you show what's broken in a screenshot?

**Reply by metacoding** [reply]
It is just not showing up
